include/realizations/catchment/Formulation_Manager.hpp

```diff
@@ -297,23 +297,100 @@ namespace realization {
                     throw std::runtime_error(message);
                 }
 
-                std::string provider = "";
-                if(forcing_parameters.has_key("provider")){
-                    provider = forcing_parameters.at("provider").as_string();
-                } else if(this->global_forcing.count("provider") != 0){
-                    provider = global_forcing.at("provider").as_string();
+                std::regex pattern("");
+                std::string path = "";
+                if(forcing_parameters.has_key("file_pattern")) {
+                    if (forcing_parameters.has_key("path")) {
+                        path = forcing_parameters.at("path").as_string();
+                    }
+
+                    // See get_global_forcing_params() function for detailed comments
+                    if (path.compare(path.size() - 1, 1, "/") != 0) {
+                        path += "/";
+                    }
+                    std::string filepattern = forcing_parameters.at("file_pattern").as_string();
+                    int id_index = filepattern.find("{{id}}");
+                    if (id_index != std::string::npos) {
+                        filepattern = filepattern.replace(id_index, sizeof("{{id}}") - 1, identifier);
+                    }
+                    std::regex pattern(filepattern);
+
+                    // A stream providing the functions necessary for evaluating a directory:
+                    //    https://www.gnu.org/software/libc/manual/html_node/Opening-a-Directory.html#Opening-a-Directory
+                    DIR *directory = nullptr;
+
+                    // structure representing the member of a directory: https://www.gnu.org/software/libc/manual/html_node/Directory-Entries.html
+                    struct dirent *entry = nullptr;
+
+                    // Attempt to open the directory for evaluation
+                    directory = opendir(path.c_str());
+
+                    std::string errMsg;
+                    size_t attemptCount = 0;
+                    std::string err_num = check_opendir(path, errMsg);
+                    if (err_num == "") {
```

Comment:
This block seems redundant, since this kind of check/retry has already been done in `check_opendir`, any error that falls through from that has already timed out or is non-recoverable, so I think the check/retry loop here can be removed.